src/lib.rs

```diff
@@ -39,25 +47,38 @@ pub struct SphinxPacket {
 
 impl SphinxPacket {
     pub fn new(
-        message: Vec<u8>,
+        mut message: Vec<u8>,
```

Comment:
Personally I don't think it's a good idea to mutate input arguments unless there's a very good reason for it